benches/language.rs

```diff
@@ -0,0 +1,259 @@
+    transforms,
+};
+
+fn benchmark_vrl_announcement(c: &mut Criterion) {
+    vector::test_util::trace_init();
+
+    let num_lines = 10_000;
+    let in_addr = next_addr();
+    let out_addr = next_addr();
+
+    let configs: Vec<(&str, config::ConfigBuilder)> = vec![
+        ("remap", {
+            let mut config = config::Config::builder();
+            config.add_source(
+                "in",
+                sources::socket::SocketConfig::make_basic_tcp_config(in_addr),
+            );
+            config.add_transform(
+                "parser",
+                &["in"],
+                toml::from_str::<transforms::remap::RemapConfig>(
+                    r#"
+source = """
+    . = parse_syslog(.message)
+    .severity = "info"
+    .id = uuid_v4()
+    .timestamp = to_int(.timestamp)
+"""
+"#,
+                )
+                .unwrap(),
+            );
+            config.add_sink(
+                "out",
+                &["parser"],
+                sinks::socket::SocketSinkConfig::make_basic_tcp_config(out_addr.to_string()),
+            );
+
+            config
+        }),
+        ("native_and_lua", {
+            let mut config = config::Config::builder();
+            config.add_source(
+                "in",
+                sources::socket::SocketConfig::make_basic_tcp_config(in_addr),
+            );
+            config.add_transform(
+                "parse_syslog",
+                &["in"],
+                toml::from_str::<transforms::regex_parser::RegexParserConfig>(
+                    r#"
+field = "message"
+patterns = ['^<(?P<priority>\d+)>(?P<version>\d+) (?P<timestamp>%S+) (?P<hostname>%S+) (?P<appname>%S+) (?P<procid>\S+) (?P<msgid>\S+) (?P<sdata>%S+) (?P<message>.+)$']
+types.appname = "string"
+types.facility = "string"
+types.hostname = "string"
+types.level = "string"
+types.message = "string"
+types.msgid = "string"
+types.procid = "int"
+types.timestamp = "timestamp|%F"
+                    "#,
+                ).unwrap(),
+            );
+            config.add_transform(
+                "add_fields",
+                &["parse_syslog"],
+                toml::from_str::<transforms::add_fields::AddFieldsConfig>(
+                    r#"
+fields.severity = "info"
+                    "#,
+                )
+                .unwrap(),
+            );
+            config.add_transform(
+                "lua_parser",
+                &["add_fields"],
+                toml::from_str::<transforms::lua::LuaConfig>(
+                    r#"
+version = "2"
+source = """
+local random = math.random
+
+local function uuid()
+    local template ='xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'
+    return string.gsub(template, '[xy]', function (c)
+        local v = (c == 'x') and random(0, 0xf) or random(8, 0xb)
+        return string.format('%x', v)
+    end)
+end
+
+function process(event, emit)
+  event.log.id = uuid()
+  event.log.timestamp = os.time(event.log.timestamp)
```

Comment:
Doesn’t the native part already take care of this?

Reply:
This is actually converting it to a unix integer timestamp.